### ISSUE TITLE
Cant run dota2 in opengl mode 

### ISSUE THREAD

**Issue by heruzek** [ValveSoftware/Dota-2#931]
Hi guys, i'm trying to fix this bug since 28/06. As you guys can see, when i set -gl, dota falls back to source shader, and says that its related to some binary.

I have already updated my drivers, and re-installed my windows several times.

A10-7700k @ 3.8
550TI (368.39)
8GB 2133mhz
W10 pro

I used to get 140~150 fps in -GL mode, but in directx i get 40 ~ 55. 
![untitled](https://cloud.githubusercontent.com/assets/20260974/16542859/4e504260-4091-11e6-8e5e-1a2c18a69522.jpg)

**Comment by gdrewb-valve** [comment]
That message is informational but things should continue on.  Does Dota not run at all after that?  What happens?

**Comment by heruzek** [comment]
Dota works well after that, but with low fps.

I don't know vconsole well, but isn't it forcing dx even when i set -gl?

"Create device: Requested DX level 110(...)" 

I am using -gl mode, and dota keeps trying to force -dx9/11.

And yes, i have already downloaded openGL dlc.

**Comment by gdrewb-valve** [comment]
The DX level does not actually mean that DX is being used, so you should be running in GL mode.  It sounds like it's just that your FPS in GL mode dropped.  Do you know roughly what time frame of update dropped your FPS?

If you turn off the Steam overlay does that change anything?  If you verify your Dota installation does that find any issues?

**Comment by heruzek** [comment]
I thought vconsole would show something related to openGL.

I have verified Dota installation and there is nothing wrong. Turned steam overlay and nothing changed.

Well, i was using W7, then i formatted to W10. (in W10 that i lost those 30~40fps)
 But if you say that i am already using -gl, its ok.

This message "Program binary could not be loaded" only shows up when i use -gl.

Sorry for any writing mistakes, english is not my native language .

**Comment by tmewett** [comment]
This tracker is for issues in the Mac and Linux client only

**Comment by danginsburg** [comment]
The odd thing here is that it should regenerate the program binary cache once it detects a failure.  Can you look in game/dota/shadercache/gl and see if the shaders.cache/shaders.toc file get updated when you run (file timestamp should change)?  If not, can you try renaming shaders.cache/shaders.toc to shaders.cache.bak/shaders.toc.bak and see if things work after running again?  If that fixes it, could you post the shaders.cache/shaders.toc files here.

Basically anytime you update drivers or hardware, the OpenGL program binaries can become no longer compatible.  However, they should be regenerated on a failure like you encountered.  I doubt this is why you have performance issues, but you should basically only be seeing that message in vconsole2 once and then it should right itself.

**Comment by heruzek** [comment]
After renaming files to .bak, i got this.
![untitled](https://cloud.githubusercontent.com/assets/20260974/16735392/40d6c8c8-475f-11e6-89b3-b7c7d8e5a131.jpg)

Also, my /gl/ folder has 4 files now.
![untitled](https://cloud.githubusercontent.com/assets/20260974/16735438/6e8f0db6-475f-11e6-885c-d322ae485130.jpg)

**Comment by gdrewb-valve** [comment]
Closing Windows issue.
